dexmaker/src/main/java/com/google/dexmaker/DexMaker.java

```diff
@@ -362,6 +410,13 @@ public ClassLoader generateAndLoad(ClassLoader parent, File dexCache) throws IOE
             }
         }
 
+        File result = new File(dexCache, generateFileName());
```

Comment:
I might be misunderstanding something, but it seems like the file caching code belongs in ProxyBuilder.  It appears that this class's purpose in life is to create dex files.  OTOH, it looks like you get to reuse some of the class loader logic that's in this class.  I'm not adamant that this needs to move, but I'd like to get a better understanding of why it should be in this class rather than ProxyBuilder.